src/clj_irods/core.clj

```diff
@@ -238,9 +258,26 @@
   [irods user zone path &
    {:keys [entity-type info-types]
     :or {entity-type :any info-types []}}] ;; only support the actual filters, we don't care about sorting and limit/offset for this
-  (let [cached (icat/filtered-cached-listings irods user zone path :entity-type entity-type :info-types info-types)]
-    (delay (get-in (first
-                     (if cached
-                       (first (icat/flatten-cached-listings @cached))
-                       (deref (icat/paged-folder-listing irods user zone path :entity-type entity-type :info-types info-types))))
-                   [:total_count]))))
+  (otel/with-span [s ["items-in-folder"]]
+    (let [cached (icat/filtered-cached-listings irods user zone path :entity-type entity-type :info-types info-types)]
+      (instrumented-delay (get-in (first
+                       (if cached
+                         (first (icat/flatten-cached-listings @cached))
+                         (deref (icat/paged-folder-listing irods user zone path :entity-type entity-type :info-types info-types))))
+                     [:total_count])))))
+
+(defn user-type
+  "The user type associated with a username. :none if the user does not exist, theoretically"
+  [irods username zone]
+  (otel/with-span [s ["user-type"]]
+    (cached-or-get irods
+      [(fn [cache? irods username zone]
+         (when-let [user (if cache?
+                           (icat/cached-user irods username zone)
+                           (icat/user irods username zone))]
+           (delay (if (:user_type_name @user) :user :none)))) username zone] ; for now mimicking jargon version that doesn't distingush
```

Comment:
I'll change this to actually specify the different user types at some point, but we don't use it right now, so I didn't for now.